qiita_pet/uimodules/prep_template_tab.py

```diff
@@ -35,13 +35,16 @@
 
 
 def _get_accessible_raw_data(user):
-    """Retrieves a tuple of raw_data_id and the last study title for that
+    """Retrieves a tuple of raw_data_id and one study title for that
     raw_data
     """
     d = {}
-    for sid in user.user_studies:
-        for rdid in Study(sid).raw_data():
-            d[int(rdid)] = Study(RawData(rdid).studies[-1]).title
+    accessible_studies = user.user_studies.union(user.shared_studies)
+    for sid in accessible_studies:
+        study = Study(sid)
+        study_title = study.title
+        for rdid in study.raw_data():
```

Comment:
Is this deployed? And, if not when?

On Sat, Jun 6, 2015 at 7:33 PM, josenavas <notifications@github.com> wrote:

> In qiita_pet/uimodules/prep_template_tab.py
> <https://github.com/biocore/qiita/pull/1246#discussion_r31873261>:
>
> >      raw_data
> >      """
> >      d = {}
> > -    for sid in user.user_studies:
> > -        for rdid in Study(sid).raw_data():
> > -            d[int(rdid)] = Study(RawData(rdid).studies[-1]).title
> > +    accessible_studies = user.user_studies.union(user.shared_studies)
> > +    for sid in accessible_studies:
> > +        study = Study(sid)
> > +        study_title = study.title
> > +        for rdid in study.raw_data():
>
> Thanks @wasade <https://github.com/wasade>
>
> I tried creating a generator that yields (rdid, study_title) and just using
>
> d = {rdid: title for rdid, title in generator}
>
> but with a real database, the accesses to the DB are so much expensive
> that the time differences due to this are minimal.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/1246/files#r31873261>.
>



-- 
Gail Ackermann
Knight Lab
UCSD
glackermann@ucsd.edu <ackermag@ucsd.edu>